### PR TITLE
WEB3#-2300: add staking api

### PR DESCRIPTION
**Pull request by yahavamar**

### CODE REVIEW

File: open_api_spec.yml [hunk 1]
```diff
@@ -1205,6 +1205,413 @@ paths:
...
+                example:
+                  - ETH
+                  - SOL
+            description: ''
+        tags:
+          - Staking (Beta)
+    /staking/chains/{chainDescriptor}/chainInfo:
+      get:
+        operationId: getChainInfo
+        summary: 'Get chain-specific, staking-related information summary'
```
Comment (dsahni) [inline open_api_spec.yml:1240]:
```suggestion
        summary: 'Get chain-specific staking summary'
```

File: open_api_spec.yml [hunk 2]
```diff
@@ -1205,6 +1205,413 @@ paths:
...
+            content:
+              application/json:
+                schema:
+                  $ref: '#/components/schemas/ChainInfoResponseDto'
+        tags:
+          - Staking (Beta)
+    /staking/chains/{chainDescriptor}/{actionId}:
+      post:
+        operationId: executeAction
+        summary: 'Execute a chain-specific staking action'
```
Comment (dsahni) [inline open_api_spec.yml:1276]:
```suggestion
        summary: 'Execute a staking action'
```

File: open_api_spec.yml [hunk 3]
```diff
@@ -1205,6 +1205,413 @@ paths:
...
+                schema:
+                  type: array
+                  items:
+                    $ref: '#/components/schemas/DelegationDto'
+        tags:
+          - Staking (Beta)
+    /staking/positions/summary:
+      get:
+        operationId: getSummaryByVaultAccountId
+        summary: 'Get staking summary details per-chain'
```
Comment (dsahni) [inline open_api_spec.yml:1406]:
```suggestion
        summary: 'Get staking summary details'
```

File: open_api_spec.yml [hunk 4]
```diff
@@ -1205,6 +1205,413 @@ paths:
...
+                          - chainDescriptor: ETH
+                            amount: '64.036604667'
+                          - chainDescriptor: SOL
+                            amount: '0.011191566'
+        tags:
+          - Staking (Beta)
+    /staking/positions/{id}:
+      get:
+        operationId: getDelegationById
+        summary: 'Get staking positions details by staking position id'
```
Comment (dsahni) [inline open_api_spec.yml:1507]:
```suggestion
        summary: 'Get staking positions details'
```

File: open_api_spec.yml [hunk 5]
```diff
@@ -1205,6 +1205,413 @@ paths:
...
+            content:
+              application/json:
+                schema:
+                  $ref: '#/components/schemas/DelegationDto'
+        tags:
+          - Staking (Beta)
+    /staking/validators/{chainDescriptor}:
+      get:
+        operationId: getValidatorsByChain
+        summary: 'List staking validators details by staking chain'
```
Comment (dsahni) [inline open_api_spec.yml:1540]:
```suggestion
        summary: 'List staking validators details'
```

File: open_api_spec.yml [hunk 6]
```diff
@@ -1205,6 +1205,413 @@ paths:
                 default:
                     $ref: "#/components/responses/Error"
 
+    # STAKING
+    /staking/chains:
+      get:
+        operationId: getChains
+        summary: 'List staking supported chains'
+        description: Return an alphabetical list of supported chains.
```
Comment (dsahni) [inline open_api_spec.yml:1213]:
```suggestion
        description: |-
          Return an alphabetical list of supported chains.  </br>
          **Note:** These endpoints are currently in beta and might be subject to changes.
```

**Review by dsahni — CHANGES_REQUESTED** [review]
1. Added suggestions to make the summary names shorter so they show up better in dev portal

2. Added an example to the first endpoint with beta language (note the multiline syntax) in `description` param. @yahavamar Can you please add similar messaging to the rest of the endpoints?

File: open_api_spec.yml (hunk 2 above)
Comment (yahavamar) [reply open_api_spec.yml]: Done

**Pull request merged**